### ISSUE TITLE
HTTP exception while notifying Bitbucket

### ISSUE THREAD

**Issue by johnyesberg** [jenkinsci/stashnotifier-plugin#232]
Hi,

Thanks for the work on this plugin.

I'm using:

* Jenkins: 2.164.3
* Windows Server 2012R2 VM
* org.jenkins-ci.plugins:stashNotifier 1.18  (known as Bitbucket Server Notifier plugin)
* BitBucket Server 5.8.1

Configuration options include:
* Bitbucket URL: `https://<bitbuckethost>/` (https and no port number, in contrast to "notifying" line in output below)
* Credential: the same SSL one that is used to retrieve builds
* Selected Options: `Ignore unverified SSL certificates`

Jenkins attempts to notify BitBucket at beginning and end of build.
Both attempts fail with an error message like this:

```
INFO: Processing Custom Tool
INFO: [Custom Tool] - 1 test report file(s) were found with the pattern 'test_results.xml' relative to 'C:\Program Files (x86)\Jenkins\workspace\conformed_accessible' for the testing framework 'Custom Tool'.
INFO: Setting the build status to SUCCESS
INFO: Stopping recording.
Build step 'Process xUnit test result report' changed build result to SUCCESS
Notifying Bitbucket at "http://<bitbuckethost>:7999"
Caught exception while notifying Bitbucket with id c90d80a1aa96d0f2e11c8dff10c205b8cadd25ab
org.apache.http.client.ClientProtocolException
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:187)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:108)
	at org.jenkinsci.plugins.stashNotifier.StashNotifier.notifyStash(StashNotifier.java:743)
	at org.jenkinsci.plugins.stashNotifier.StashNotifier.processJenkinsEvent(StashNotifier.java:342)
	at org.jenkinsci.plugins.stashNotifier.StashNotifier.perform(StashNotifier.java:298)
	at org.jenkinsci.plugins.stashNotifier.StashNotifier.perform(StashNotifier.java:257)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:744)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:690)
	at hudson.model.Build$BuildExecution.post2(Build.java:186)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:635)
	at hudson.model.Run.execute(Run.java:1841)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
Caused by: org.apache.http.ProtocolException: The server failed to respond with a valid HTTP response
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:149)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:56)
	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:259)
	at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:163)
	at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:165)
	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:273)
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:125)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:272)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
	... 15 more
Finished: SUCCESS
```

**Comment by scaytrase** [comment]
while the exception looks ok, I see the following line

```
Caused by: org.apache.http.ProtocolException: The server failed to respond with a valid HTTP response
```

Are you sure that jenkins master can correctly communicate with bitbucket? Cause says that jenkins tried notification, but bitbucked (or something between them) gave it wrong response

**Comment by johnyesberg** [comment]
@scaytrase, thanks for your comment.
I made a mistake in my original post. The Bitbucket Status Notifier configuration doesn't use the same SSH credentials: it uses username and password credentials for the same account. I had been using the wrong password. When I fixed the password, the error changed. It think it is appropriate to close this issue and if I can't solve the new SSL related issue (perhaps certificate-related), I will open a new issue.
